pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -602,24 +602,22 @@ def view_queues():
         else:
             return None
 
-    def to_dict(self):
-        server_dict = self._type_to_dict()
-        server_dict.update(
-            {
-                "user": self._user,
-                "host": self._host,
-                "run_mode": self.run_mode.mode,
-                "queue": self.queue,
-                "qid": self._queue_id,
-                "cores": self.cores,
-                "threads": self.threads,
-                "new_h5": self.new_hdf,
-                "structure_id": self.structure_id,
-                "run_time": self.run_time,
-                "memory_limit": self.memory_limit,
-                "accept_crash": self.accept_crash,
-            }
-        )
+    def _to_dict(self):
+        # server_dict = self._type_to_dict()
+        server_dict = {
+            "user": self._user,
+            "host": self._host,
+            "run_mode": self.run_mode.mode,
+            "queue": self.queue,
+            "qid": self._queue_id,
+            "cores": self.cores,
+            "threads": self.threads,
+            "new_h5": self.new_hdf,
+            "structure_id": self.structure_id,
+            "run_time": self.run_time,
+            "memory_limit": self.memory_limit,
+            "accept_crash": self.accept_crash,
+        }
```

Comment:
In analogy to the `Executable` class, does the `Server` class need the transition from `to_dict()` to `_to_dict()` or can we modify the class in a way that the `__getstate__()` returns the same dictionary as `to_dict()`. Again my focus would be on simplifying the class hierarchy.

Reply:
Both Server and Executable derived from `HasDict` before this PR.  I'm ok with changing this in the future, but here I just did the minimal changes to keep existing sub classes working with the changes I did to `HasDict` (which is just adding the type info automatically to the results of to_dict).

Reply:
I add this to the bucket list. I would like to replace both with data classes sooner rather than later then we can potentially also reuse them in the functional approach.